pandas/tests/io/test_sql.py

```diff
@@ -2909,17 +2910,18 @@ def test_datetime(conn, request):
     # with read_table -> type information from schema used
     result = sql.read_sql_table("test_datetime", conn)
     result = result.drop("index", axis=1)
-    tm.assert_frame_equal(result, df)
+
+    expected = df[:]
```

Comment:
Instead of changing this here does the test work if you add usec precision to the string passed into `date_range`? AFAIK using usec precision for storage of dates / datetimes is relatively common (at least in the postgres family) so would be fine to just start with that on writing instead of munging the expectation